cmds/core/truncate/truncate.go

```diff
@@ -34,50 +36,38 @@ var (
 
 func init() {
 	flag.Var(size, "s", "Size in bytes, prefixes +/- are allowed")
-
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
-	}
-}
-
-func usageAndExit() {
-	flag.Usage()
-	os.Exit(1)
+	util.Usage(usage)
 }
 
-func main() {
-	flag.Parse()
-
+func truncate(args ...string) error {
 	if !size.IsSet && *rfile == "" {
-		log.Println("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>.")
-		usageAndExit()
+		flag.Usage()
+		return fmt.Errorf("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>")
 	}
 	if size.IsSet && *rfile != "" {
-		log.Println("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>.")
-		usageAndExit()
+		flag.Usage()
+		return fmt.Errorf("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>")
 	}
-	if flag.NArg() == 0 {
-		log.Println("truncate: ERROR: You need to specify one or more files as argument.")
-		usageAndExit()
+	if len(args) == 0 {
+		flag.Usage()
+		return fmt.Errorf("truncate: ERROR: You need to specify one or more files as argument")
 	}
 
-	for _, fname := range flag.Args() {
+	for _, fname := range args {
 
 		var final int64
 		st, err := os.Stat(fname)
 		if os.IsNotExist(err) && !*create {
 			if err = os.WriteFile(fname, []byte{}, 0o644); err != nil {
-				log.Fatalf("truncate: ERROR: %v\n", err)
+				return fmt.Errorf("truncate: ERROR: %v", err)
```

Comment:
I think you might take out the ERROR: part of these error messages? 

Also in the standard command create is true.
Check out this bizarre flag:
       -c, --no-create
              do not create any files

It gets better: 
rminnich@t510:~$ truncate -c -s  0 /tmp/asdfadfadsffd
rminnich@t510:~$ rm !$
rm /tmp/asdfadfadsffd
rm: cannot remove '/tmp/asdfadfadsffd': No such file or directory
rminnich@t510:~$

notice: no file created, no error message. Awesome! What should *we* do? At least, I think, make the flag true.

Reply:
Regarding this https://man7.org/linux/man-pages/man1/truncate.1.html the flag create is correct as it is. If you set the -c flag than no file will be created.